### PR TITLE
S1 et4

### PR DESCRIPTION
**Pull request by AlisonYoon**

### CODE REVIEW

File: function.php [hunk 1]
```diff
@@ -1,24 +1,107 @@
 <?php
 require_once 'db.php';
 
-function getData($db) {
+/**
+ * getData function gets data from the DB
+ * @param $db
+ * @return mixed
```
Comment (FelixLetheren) [inline function.php]: docbloc says mixed when it is type hinted to array

File: function.php [hunk 2]
```diff
@@ -1,24 +1,107 @@
...
+    $sql = $db->prepare($request);
     $sql->execute();
     $groceryItems = $sql->fetchAll();
 
     return $groceryItems;
 }
 
-function processData($groceryItems) {
+/**
+ * generateRequest function was created for unit testing, it took some part from getData function
```
Comment (FelixLetheren) [inline function.php]: doesn't really describe what function actually does

File: test/testFunction.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    public function testGenerateRequestShouldReturnWhereCategoryWithParam()
+    {
+        //setup
+        $filter = 'pantry';
+        $result = generateRequest($filter);
+        $expectedResult = 'SELECT `item`, `category`, `price`, `remaining` FROM `grocery_item` WHERE `category` = "' . $filter.  '";';
+
+        //assertion
+        $this->assertEquals($result, $expectedResult);
```
Comment (FelixLetheren) [inline test/testFunction.php:26]: Could maybe do with failure and malformed data unit tests as well? type error etc.?

**Review by FelixLetheren — CHANGES_REQUESTED** [review]

File: function.php (hunk 1 above)
Comment (AlisonYoon) [reply function.php]: ok fixed

File: function.php (hunk 2 above)
Comment (AlisonYoon) [reply function.php]: good point

File: test/testFunction.php (hunk 3 above)
Comment (AlisonYoon) [reply test/testFunction.php:26]: did malformed unit test! thanks

**Pull request merged**